extensions/amp-story/0.1/amp-story-page.js

```diff
@@ -0,0 +1,642 @@
+const TAG = 'amp-story-page';
+
+
+
+export class AmpStoryPage extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private @const {?AnimationManager} */
+    this.animationManager_ = null;
+
+    /** @private @const {!Array<!PageElement>} */
+    this.pageElements_ = [];
+
+    /** @private {?function()} */
+    this.resolveLoadPromise_ = null;
+
+    /** @private {?Promise<undefined>} */
+    this.loadPromise_ = new Promise(resolve => {
+      this.resolveLoadPromise_ = resolve;
+    });
+
+    /** @private {?Promise<undefined>} */
+    this.loadTimeoutPromise_ = null;
+
+    /** @private @const {!../../../src/service/timer-impl.Timer} */
+    this.timer_ = Services.timerFor(this.win);
+
+    /** @private {boolean} */
+    this.isLoaded_ = false;
+
+    /** @private {?UnlistenDef} */
+    this.autoAdvanceUnlistenDef_ = null;
+  }
+
+
+  /*
+   * @return {?./animation.AnimationManager}
+   */
+  maybeCreateAnimationManager_() {
+    if (!this.animationManager_) {
+      if (!hasAnimations(this.element)) {
+        return;
+      }
+
+      this.animationManager_ = AnimationManager.create(
+          this.element, this.getAmpDoc(), this.getAmpDoc().getUrl());
+    }
+  }
+
+
+  /** @override */
+  buildCallback() {
+    upgradeBackgroundAudio(this.element);
+    this.markMediaElementsWithPreload_();
+    this.maybeCreateAnimationManager_();
+    this.initializeLoading_();
+  }
+
+
+  /**
+   * Marks any AMP elements that represent media elements with preload="auto".
+   */
+  markMediaElementsWithPreload_() {
+    const mediaSet = this.element.querySelectorAll('amp-audio, amp-video');
+    for (const mediaItem of mediaSet) {
+      mediaItem.setAttribute('preload', 'auto');
+    }
+  }
+
+
+  /**
+   * Initializes the loading screen for this amp-story-page, and the listeners
+   * to remove it once loaded.
+   */
+  initializeLoading_() {
+    // Add the loading screen into the DOM.
+    const loadingScreen = this.win.document.createElement('div');
+    loadingScreen.classList.add('i-amphtml-story-page-loading-screen');
+    loadingScreen./*OK*/innerHTML = LOADING_SCREEN_CONTENTS_TEMPLATE;
+    this.element.appendChild(loadingScreen);
+
+    // Build a list of page elements and poll until they are all loaded.
+    this.pageElements_ = PageElement.getElementsFromPage(this);
+    this.loadPromise_ = this.timer_.poll(LOAD_TIMER_POLL_DELAY_MS, () => {
+      return this.calculateLoadStatus();
+    }).then(() => this.markPageAsLoaded_());
+  }
+
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.CONTAINER;
+  }
+
+
+  /** @override */
+  pauseCallback() {
+    this.pageInactiveCallback_();
+  }
+
+
+  /** @override */
+  resumeCallback() {
+    this.pageActiveCallback_();
+  }
+
+
+  /** @private */
+  onPageVisible_() {
+    this.markPageAsLoaded_();
+    this.maybeApplyFirstAnimationFrame();
+    this.maybeScheduleAutoAdvance_();
```

Comment:
Minor: A lot of "maybe" prefixes start to lose meaning -- consider removing or making more specific.